### DISCUSSION TITLE
OpenModelica uninstaller removed all my files

### DISCUSSION THREAD

**Discussion by tanushree04** [OpenModelica/OpenModelica#12625] [General]
Hi, 
I tried uninstalling OpenModelica for this link https://build.openmodelica.org/omc/builds/windows/releases/1.22/1/64bit/
And the uninstaller removed more than 90% of the files from my user profile from desktop, documents, c drive. While it was doing so I could see in the log it was looping through all my files and removing them but didn’t want to interrupt and trusted the process assuming there were some modelica related files there that it was trying to remove. These files can not be found anywhere IT cannot tell what happened to them. Or why the uninstaller instead of removing program specific files removed my files at large. The command in the computer log was something along the lines of an ‘admin burn’. My computer is now being sent for professional data recovery. I would appreciate any help, insights on what could have happened as this will help with the retrieval  process 
Thank you

**Comment by casella** [comment]
@adrpo this is strange, any clue?

@tanushree04 I would always recommend to use the latest released version, in this case 1.23.0. We strive hard to keep full backwards compatibility, so in general there is no point using older versions of the software.

**Comment by adeas31** [comment]
The only possibility of it that comes to mind is that if you install in `C:\`.

I will take a look in then installer source code on how the uninstall is done.

**Reply by AnHeuermann** [reply]
Doesn't seem to be possible to install in `C:\`, but you can choose `C:\Users\<username>` with the v1.23.0 installer. And indeed it will happily purge my user directory.

![grafik](https://github.com/OpenModelica/OpenModelica/assets/38031952/f22b599e-901c-4c7e-b0fc-fe966310ad1d)

**Comment by casella** [comment]
@adeas31 is working on it, to make sure only installed files will be removed.

**Comment by adeas31** [comment]
Now the installer only removes the installed files. See https://github.com/OpenModelica/OpenModelicaSetup/pull/62
